specification/developersigning/DeveloperSigning/main.tsp

```diff
@@ -0,0 +1,196 @@
+  RS256,
+
+  @doc("RSASSA-PKCS1-v1_5 using SHA-384 hash algorithm.")
+  RS384,
+
+  @doc("RSASSA-PKCS1-v1_5 using SHA-512 hash algorithm.")
+  RS512,
+
+  @doc("RSASSA-PSS using SHA-256 hash algorithm.")
+  PS256,
+
+  @doc("RSASSA-PSS using SHA-384 hash algorithm.")
+  PS384,
+
+  @doc("RSASSA-PSS using SHA-512 hash algorithm.")
+  PS512,
+
+  @doc("ECDSA using P-256 and SHA-256 hash algorithm.")
+  ES256,
+
+  @doc("ECDSA using P-384 and SHA-384 hash algorithm.")
+  ES384,
+
+  @doc("ECDSA using P-521 and SHA-512 hash algorithm.")
+  ES512,
+
+  @doc("ECDSA using secp256k1 and SHA-256 hash algorithm.")
+  ES256K,
+}
+
+@doc("Azure Developer Signing customer account created upon registration on the service.")
+@resource("codesigningaccounts")
+model SigningAccount {
+  @key("accountName")
+  @visibility("read", "update")
+  @doc("Azure Developer Signing account name.")
+  accountName: string;
+}
+
+@doc("Azure Developer Signing certificate profile configured on an account.")
+@resource("certificateprofiles")
+@parentResource(SigningAccount)
+model CertificateProfileName {
+  @key("certificateProfile")
+  @visibility("read", "update")
+  @doc("Azure Developer Signing certificate profile name under an account.")
+  certificateProfile: string;
+}
+
+@doc("The artifact request information to be signed by the service.")
+model SigningPayloadOptions {
+  @doc("The supported signature algorithm identifiers.")
+  signatureAlgorithm: SignatureAlgorithm;
+
+  @doc("Content digest to sign.")
+  digest: bytes;
+
+  @doc("List of full file digital signatures.")
+  fileHashList?: bytes[];
+
+  @doc("List of authenticode digital signatures.")
+  authenticodeHashList?: bytes[];
+}
+
+@doc("The sign status model.")
+model SignResult {
+  @doc("Digital signature of the requested content digest.")
+  signature?: bytes;
+
+  @doc("""
+Signing certificate corresponding to the private key used to sign the requested 
+digest.
+""")
+  signingCertificate?: bytes;
+}
+
+@doc("Signing option parameters corresponding to the account and certificate profile name.")
+model SigningOptions {
+  @doc("Signing account name.")
+  @path
+  accountName: string;
+
+  @doc("Signing Certificate profile name.")
+  @path
+  certificateProfile: string;
+}
+
+@doc("Extended key usage object identifier that are allowable.")
+@resource("sign/eku")
+@parentResource(CertificateProfileName)
+model ExtendedKeyUsage {
+  @key("eku")
+  @visibility("read")
+  @doc("An oid string that represents an eku.")
+  eku: string;
+}
+
+@doc("Public root certificate from the certificate chain.")
+model BytesBody<ContentType> {
+  @doc("The root certificate of the certificate chain of this profile.")
+  @body
+  body: bytes;
+
+  @doc("The content type of the x509 cert.")
+  @header
+  contentType: ContentType;
+}
+
+alias SigningRequestHeadersTraits = Azure.Core.Traits.RequestHeadersTrait<{
+  @doc("An optional client version.")
+  @header
+  clientVersion: string;
```

Comment:
If this header is optional you must indicate that with a "?" after the name.
```suggestion
  clientVersion?: string;
```

Reply:
Thanks, this is fixed on the PR.